src/test/kotlin/pl/edu/agh/gem/internal/service/ExpenseServiceTest.kt

```diff
@@ -0,0 +1,255 @@
+import org.mockito.kotlin.anyVararg
+import org.mockito.kotlin.atLeastOnce
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import pl.edu.agh.gem.helper.group.DummyGroup.GROUP_ID
+import pl.edu.agh.gem.helper.group.createGroupMembers
+import pl.edu.agh.gem.helper.user.DummyUser.OTHER_USER_ID
+import pl.edu.agh.gem.helper.user.DummyUser.USER_ID
+import pl.edu.agh.gem.internal.client.CurrencyManagerClient
+import pl.edu.agh.gem.internal.client.GroupManagerClient
+import pl.edu.agh.gem.internal.model.currency.Currencies
+import pl.edu.agh.gem.internal.model.expense.Expense
+import pl.edu.agh.gem.internal.persistence.ExpenseRepository
+import pl.edu.agh.gem.util.DummyData.CURRENCY_1
+import pl.edu.agh.gem.util.DummyData.CURRENCY_2
+import pl.edu.agh.gem.util.createCurrencies
+import pl.edu.agh.gem.util.createExchangeRate
+import pl.edu.agh.gem.util.createExpense
+import pl.edu.agh.gem.util.createExpenseParticipant
+import pl.edu.agh.gem.util.createGroupOptions
+import pl.edu.agh.gem.validator.ValidatorsException
+import java.math.BigDecimal
+import java.time.Instant
+
+class ExpenseServiceTest : ShouldSpec({
+    val groupManagerClient = mock<GroupManagerClient> { }
+    val currencyManagerClient = mock<CurrencyManagerClient> {}
+    val expenseRepository = mock<ExpenseRepository> {}
+    val expenseService = ExpenseService(
+        groupManagerClient = groupManagerClient,
+        currencyManagerClient = currencyManagerClient,
+        expenseRepository = expenseRepository,
+    )
+    should("get group members from client") {
+        // given
+        val groupMembers = createGroupMembers(listOf(USER_ID, OTHER_USER_ID))
+        whenever(groupManagerClient.getGroupMembers(GROUP_ID)).thenReturn(groupMembers)
+
+        // when
+        val result = expenseService.getGroupMembers(GROUP_ID)
+
+        // then
+        verify(groupManagerClient, atLeastOnce()).getGroupMembers(GROUP_ID)
+        result shouldBe groupMembers
+    }
+
+    should("create expense") {
+        // given
+        val expense = createExpense()
+        val groupMembers = createGroupMembers(listOf(USER_ID, OTHER_USER_ID))
+        val exchangeRate = createExchangeRate()
+        val expected = expense.copy(exchangeRate = exchangeRate)
+        whenever(groupManagerClient.getGroupOptions(GROUP_ID)).thenReturn(createGroupOptions(currencies = createCurrencies(CURRENCY_1, CURRENCY_2)))
+        whenever(currencyManagerClient.getAvailableCurrencies()).thenReturn(createCurrencies(CURRENCY_1, CURRENCY_2))
+        whenever(currencyManagerClient.getExchangeRate(CURRENCY_1, CURRENCY_2, Instant.ofEpochMilli(0L))).thenReturn(exchangeRate)
+        whenever(expenseRepository.create(anyVararg(Expense::class))).thenReturn(expected)
+
+        // when
+        val result = expenseService.create(groupMembers, expense)
+
+        // then
+        result shouldNotBe null
+        result.also {
```

Comment:
if "result" is the same type as "expeted" should we just result shoulBe expected ?